### PR TITLE
Clarify pub.dart messages

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
cc @yjbanov

**Comment by Hixie** [comment]
cc @devoncarew

File: packages/flutter_tools/lib/src/dart/pub.dart [hunk 1]
```diff
@@ -58,20 +58,22 @@ Future<Null> pubGet({
...
 
-  throwToolExit('$directory: pubspec.yaml and .packages are in an inconsistent state');
+  if (dotPackages.lastModifiedSync().isBefore(pubSpecYaml.lastModifiedSync()))
+    throwToolExit('$directory: pub did not update .packages file (pubspec.yaml file has a newer timestamp)');
 }
 
-String _filterOverrideWarnings(String str) {
-  // Warning: You are using these overridden dependencies:
-  // ! analyzer 0.29.0-alpha.0 from path ../../bin/cache/dart-sdk/lib/analyzer
+final RegExp _analyzerWarning = new RegExp(r'^! analyzer [^ ]+ from path \.\./\.\./bin/cache/dart-sdk/lib/analyzer$');
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/dart/pub.dart]: Is the relative path always the same? For example, if the analyzer override transitive, would it still produce the same path?

**Review by yjbanov — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/dart/pub.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/dart/pub.dart]: Seems to. I tried creating a package way off in a different directory, then ran `flutter packages get` and it didn't print anything.

**Pull request merged**